cli/pkg/local/server.go

```diff
@@ -90,6 +103,394 @@ func (s *Server) GetVersion(ctx context.Context, r *connect.Request[localv1.GetV
+			IsGithubRepo:                  false,
+			IsGithubRemoteFound:           false,
+			IsGithubRepoAccessGranted:     false,
+			GithubUrl:                     "",
+			HasUncommittedChanges:         nil,
+			RillOrgExistsAsGithubUserName: false,
+			RillUserOrgs:                  nil,
+			LocalProjectName:              localProjectName,
+		}), nil
+	}
+
+	isGithubRepo := true
+	githubRemoteFound := true
+	repoAccess := false
+	// check if project is a git repo
+	remote, ghURL, err := gitutil.ExtractGitRemote(s.app.ProjectPath, "", false)
+	if err != nil {
+		if errors.Is(err, git.ErrRepositoryNotExists) {
+			isGithubRepo = false
+		} else if errors.Is(err, gitutil.ErrGitRemoteNotFound) {
+			githubRemoteFound = false
+		} else {
+			return nil, err
+		}
+	}
+
+	var hasUncommittedChanges *bool
+	if isGithubRepo && githubRemoteFound {
+		// check if there are uncommitted changes
+		// ignore errors since check is best effort and can fail in multiple cases
+		syncStatus, _ := gitutil.GetSyncStatus(s.app.ProjectPath, "", remote.Name)
+		hasUncommittedChanges = new(bool)
+		if syncStatus == gitutil.SyncStatusModified || syncStatus == gitutil.SyncStatusAhead {
+			*hasUncommittedChanges = true
+		} else {
+			*hasUncommittedChanges = false
+		}
+
+		// check git repo status, if git app installed and user is a collaborator and has authorised app on their account
+		repoStatus, err := c.GetGithubRepoStatus(ctx, &adminv1.GetGithubRepoStatusRequest{
+			GithubUrl: ghURL,
+		})
+		if err != nil {
+			return nil, err
+		}
+		repoAccess = repoStatus.HasAccess
+		if !repoStatus.HasAccess {
+			// we should have access to the repo before deploying
+			return connect.NewResponse(&localv1.DeployValidationResponse{
+				IsAuthenticated:         true,
+				LoginUrl:                loginURL,
+				IsGithubConnected:       true,
+				GithubGrantAccessUrl:    userStatus.GrantAccessUrl,
+				GithubUserName:          userStatus.Account,
+				GithubAppUserPermission: userStatus.UserInstallationPermission,
+				GithubOrganizationInstallationPermissions: userStatus.OrganizationInstallationPermissions,
+				IsGithubRepo:                  true,
+				IsGithubRemoteFound:           true,
+				IsGithubRepoAccessGranted:     repoAccess,
+				GithubUrl:                     "",
+				HasUncommittedChanges:         hasUncommittedChanges,
+				RillOrgExistsAsGithubUserName: false,
+				RillUserOrgs:                  nil,
+				LocalProjectName:              localProjectName,
+			}), nil
+		}
+	}
+
+	// get rill user orgs
+	resp, err := c.ListOrganizations(ctx, &adminv1.ListOrganizationsRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	userOrgs := make([]string, 0, len(resp.Organizations))
+	for _, org := range resp.Organizations {
+		userOrgs = append(userOrgs, org.Name)
+	}
+	// TODO if len(userOrgs) > 0 then check if any project in these orgs already deploys from ghUrl, however I think this is not enough, probably should be checked globally ?
```

Comment:
Is this important to address?

Reply:
cli flow checks this [here](https://github.com/rilldata/rill/blob/main/cli/cmd/deploy/deploy.go#L285) but it only check in the current org, may be its fine as current user might not have access to another org. There 3 actions that user can take, 2 should be taken from cli and one is continue deployment with another project name, so was not sure how this should be surfaced so did not added this validation. Can be a follow up PR, may be just add flag to indicate this and UI can just say - `Another project deploy from this repo, to abort press abort or continue with another name` ?